Cygwin: Force keyboard state onto VCK at startup

Copy the state of the Windows keyboard device to the Virtual Core Keyboard at startup.

Otherwise, this happens lazily after the first keypress, which can lead to applications
which are started from a shell window and inspect the keyboard state before a character is
typed getting the wrong idea about the desired keymap (e.g. xemacs shows this behaviour)

---
 xserver/hw/xwin/winkeybd.c |    6 ++++++
 1 file changed, 6 insertions(+)

 InitInput (int argc, char *argv[])
Index: xorg-server-1.6.0/xserver/hw/xwin/winkeybd.c
===================================================================
--- xorg-server-1.6.0.orig/xserver/hw/xwin/winkeybd.c	2009-02-26 22:36:39.718750000 +0000
+++ xorg-server-1.6.0/xserver/hw/xwin/winkeybd.c	2009-02-26 22:36:40.375000000 +0000
@@ -288,12 +288,18 @@ winKeybdProc (DeviceIntPtr pDeviceInt, i
             {  
               winErrorFVerb (1, "winKeybdProc - Error initializing keyboard AutoRepeat (No XKB)\n");
             }
+
+	  XkbSetExtension(pDeviceInt, ProcessKeyboardEvent);
         }
 #endif
       break;
       
     case DEVICE_ON: 
       pDevice->on = TRUE;
+
+      // immediately copy the state of this keyboard device to the VCK
+      // (which otherwise happens lazily after the first keypress)
+      CopyKeyClass(pDeviceInt, inputInfo.keyboard);
       break;
 
     case DEVICE_CLOSE:
